lecture_1/cli_args.js

```diff
@@ -1,26 +1,46 @@
-const fs = require("fs");
 const { stat } = require("fs/promises");
+const yargs = require("yargs/yargs");
+const { hideBin } = require("yargs/helpers");
 
-const filePath = process.argv[2];
+const PATH_FLAG = "path";
+const ENTITY_TYPE = "entity-type";
 
-console.log(`Getting stats of ${filePath}`);
+const argv = yargs(hideBin(process.argv)).argv;
 
-// fs.stat(filePath, (err, stats) => {
-//   if (err) {
-//     throw err;
-//   }
+if (!argv[PATH_FLAG]) {
+  console.log("Error: path is not defined");
+  process.exit();
+}
 
-//   console.log(stats);
-// });
+console.log(`Getting stats of ${argv[PATH_FLAG]}`);
 
 (async () => {
   try {
-    const metaData = await stat(filePath);
-
+    const metaData = await stat(argv[PATH_FLAG]);
     console.log(metaData);
+    if (argv[ENTITY_TYPE]) {
+      console.log(`type: ${getType(metaData)}`);
+    }
   } catch (err) {
     console.error(err);
-
     process.exitCode = 1;
   }
 })();
+
+function getType(stats) {
+  const types = new Map([
+    ["file", stats.isFile()],
```

Comment:
It's better to invoke those stats.* methods lazy and immediately return type if first of them return true.
The solution with map and forEach is a bit over-engineered overall=).